website/content/specification/_index.md

```diff
@@ -1,50 +1,1320 @@
+
+An example a collapsible field definition might look like the following.
+
+```xml
+<define-assembly name="assembly">
+  <root-name>assembly</root-name>
+  <model>
+    <define-field name="field" max-occurs="unbounded" collapsible="yes">
+      <group-as name="fields" in-json="ARRAY"/>
+      <define-flag name="flag-default-a" default="a"/>
+      <define-flag name="flag-required" required="yes"/>
+      <define-flag name="flag-optional"/>
+    </define-field>
+  </model>
+</define-assembly>
+```
+
+An example set of content instances follow. XML is provided first to illustrate the use of uncollapsed fields, since this feature applies to JSON and YAML only. The JSON and YAML collapsed versions follow.
+
+```xml {linenos=table,hl_lines=[2,4]}
+<assembly>
+  <field flag-required="required 1">field-value-1</field>
+  <field flag-required="required 2">field-value-2</field>
+  <field flag-default-a="a" flag-required="required 1">field-value-3</field>
+</assembly>
+```
+
+The XML example above illustrates 2 fields, on lines 2 and 4, which will be collapsed together into the same *collapse group* in the resulting JSON and YAML conversion. This is because they have the same flag values: `a` for the flag named `flag-default-a`, `required 1` for the flag named `flag-required`, and no value for the flag named `flag-optional`. The field on line 3 would be in a second *collapse group*.
+
+When converted to JSON or YAML, these fields will be collapsed based on their collapse grouping, resulting in the following content instances.
+
+{{< tabs JSON YAML >}}
+{{% tab %}}
+```json {linenos=table,hl_lines=["4-7"]}
+{
+"assembly": {
+  "fields": [
+    {
+      "flag-required": "required 1",
+      "STRVALUE": [ "field-value-1", "field-value-3"]
+    },
+    {
+      "flag-required": "required 2",
+      "STRVALUE": "field-value-2"
+    }
+  ]
+}
+```
+{{% /tab %}}
+{{% tab %}}
+```yaml {linenos=table,hl_lines=["4-7"]}
+---
+assembly:
+  fields:
+  - flag-required: "required 1"
+    STRVALUE:
+    - "field-value-1"
+    - "field-value-3"
+  - flag-required: "required 2"
+    STRVALUE: "field-value-2"
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+If the JSON or YAML instance is converted back to XML, the sequencing of the fields will change due to use of the collapsible feature as follows.
+
+```xml {linenos=table,hl_lines=["2-3"]}
+<assembly>
+  <field flag-required="required 1">field-value-1</field>
+  <field flag-required="required 1">field-value-3</field>
+  <field flag-required="required 2">field-value-2</field>
+</assembly>
+```
+
+In the example above, the fields from the first *collapse group* would appear before the fields from the second *collapse group*. This is because the resulting JSON and YAML instance has no way of indicating the original sequencing, so it must rely on the sequencing provided by the ordering of the *collapse groups* and the values within each group.
+
+### `@default`
+
```

Comment:
Did we discuss this feature? Where is it used?

Do we provide advice to implementers on how to process Metaschema-described data in the absence of schemas or schema bindings providing normative defaults? (I can imagine an XSLT that would provide such defaults but that's XSLT.)

(In OSCAL we have a nominal default for the `@ns` flag, but for no fields tmk?)

Reply:
@wendellpiez We did a while back. You had mixed feelings about this feature.

This allows for the declaration of the nominal default.

I'd like to keep schema bindings out of the discussion here. This would be discussed in specs around schema generation, which is out of scope here. Given this, what additional advice would you like to include here?